graylog2-server/src/main/java/org/graylog/events/context/EventDefinitionContextService.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.events.context;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableMap;
+import org.graylog.events.processor.EventDefinitionDto;
+import org.graylog.events.processor.EventProcessorExecutionJob;
+import org.graylog.scheduler.DBJobDefinitionService;
+import org.graylog.scheduler.DBJobTriggerService;
+import org.graylog.scheduler.JobDefinitionDto;
+import org.graylog.scheduler.JobTriggerData;
+import org.graylog.scheduler.JobTriggerDto;
+import org.graylog.scheduler.JobTriggerStatus;
+import org.joda.time.DateTime;
+
+import javax.inject.Inject;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class EventDefinitionContextService {
+    private final DBJobDefinitionService jobDefinitionService;
+    private final DBJobTriggerService jobTriggerService;
+
+    @Inject
+    public EventDefinitionContextService(DBJobDefinitionService jobDefinitionService, DBJobTriggerService jobTriggerService) {
+        this.jobDefinitionService = jobDefinitionService;
+        this.jobTriggerService = jobTriggerService;
+    }
+
+    public ImmutableMap<String, Object> contextFor(List<EventDefinitionDto> eventDefinitions) {
+        return ImmutableMap.of("scheduler", schedulerContext(eventDefinitions));
+    }
+
+    public ImmutableMap<String, Object> contextFor(EventDefinitionDto eventDefinition) {
+        final ImmutableMap<String, SchedulerCtx> schedulerContext = schedulerContext(Collections.singletonList(eventDefinition));
+        return ImmutableMap.of("scheduler", schedulerContext.get(eventDefinition.id()));
+    }
+
+    private Map<String, List<JobDefinitionDto>> getJobDefinitions(List<EventDefinitionDto> eventDefinitions) {
+        final Set<String> eventDefinitionIds = eventDefinitions.stream().map(EventDefinitionDto::id).collect(Collectors.toSet());
+        return jobDefinitionService.getAllByConfigField(EventProcessorExecutionJob.Config.FIELD_EVENT_DEFINITION_ID, eventDefinitionIds);
```

Comment:
If we want to be extra correct, we would also query for the type being `EventProcessorExecutionJob.TYPE_NAME`

Reply:
We this in several other places as well without the type. I think it wouldn't be possible that we have two event definitions with the same `_id` but different types. Since the `_id` has a unique index.

Reply:
My concern was not duplicates, but the fact that we are the config document from other types doesn't have the field we are querying.
But I guess it's fine.

Reply:
Hmm. But then we would just get no result from the query, no?

Reply:
yeah, unless someone creates a plugin that uses the same field name in its config, which is quite unlikely :-D